### PR TITLE
Fix #306: Custom elements in page containers

### LINKED ISSUE #306: SeLion-Code-Generator generates custom element incorrectly when defined in a Container
I am experiencing weird behavior when a custom HTML element is defined within a Container of a PageYAML file.  The resulting generated code incorrectly generates the getter for that element at the Page level instead of the Container level.  The object mapping for the element exists within the Container mapping (as expected).  Calling the getter at the Page level (as currently generated) results in a null locator being passed to the custom element, making that element unusable in the current implementation.  What configuration is needed for it to generate within the Container?

### PR DESCRIPTION
**Pull request by sebady**
- [x] By placing an `X` in the preceding checkbox, I verify that I am a PayPal employee or
I have signed the [Contributor License Agreement](https://github.com/paypal/SeLion#contributing)

---

- Fix #306
- Support custom elements in page object containers
- Also unrelated change in parent POM to remove
  eclipse lifecycle mapping plugin.

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -160,35 +160,6 @@
     <build>
         <pluginManagement>
             <plugins>
-                <!-- this plugin is a maven plugin designed to hold meta data to communicate with eclipse m2e plugin when 
-                    it processes a maven pom.xml. Without this, children of this pom file will get the warning 
-                    "maven-enforcer-plugin (goal enforce) is ignored by m2e" because there is no connector that tells 
-                    m2e how to integrate the maven plugin into eclipse. -->
-                <plugin>
```
Comment (mach6) [inline pom.xml:167]: Not needed anymore?

**Comment by mach6** [comment]
Is there a test that can be added to ensure the bug doesn't resurface?

File: pom.xml (hunk 1 above)
Comment (sebady) [reply pom.xml:167]: Recent m2e versions don't need this anymore and I don't believe any recent version of eclipse _requires_ this now. I know for certain eclipse (neon) builds SeLion fine w/o this.  For users still on old versions of Eclipse (for some reason) the recommendation is to add this to ~/.m2/settings.xml

Comment (mach6) [reply pom.xml:167]: good to learn and good catch!

**Review by mach6 — APPROVED** [review]
This looks good. Just need the test(s).

**Comment by mach6** [comment]
merged this to 1.2.0

**Pull request closed without merging**